BSDLib in Zephyr without TrustZone

sigvartmh opened the pull request:
I do know that it's a possibility we don't want this in nRF hence pointing my PR towards my own public fork. I just think it's cool to allow customers to use the chip to it's full extent without being weighed down by trustZone and having their RAM partitioned.

I'm not sure if the linker script generated is correct yet. Since I haven't hit the boundary between BSDLib and end of BSDLib but the general idea is too use a statically allocated block of memory which is placed in the `.bsdlib` section which is located in the `BSD_RESERVED_MEMORY_ADDRESS` and uses the `BSD_RESERVED_MEMORY_SIZE` to fit into the region.

Further more when initializing the BSD Library we calculate which regions needs to be marked non-secure through using those defines and the `NRF_SPU_S`. 

One advantage of doing this is that you'll be able to access CryptoCell functionality directly. Meaning you'll have an easier time using CryptoCell+mbedTLS+raw sockets to create a new TLS/DTLS session additionally to the one you have available in the modem.

oyvindronningstad committed:
secure_services: Add secure service for prevalidation of B1 upgrades

The service calls through to B0.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

oyvindronningstad committed:
fw_info: Kconfig: Move one config and remove some prompts

Move one config should have been moved to the bootloader.
Remove prompt on some configs that should not be changed.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

oyvindronningstad committed:
bl_validation: Simplify OFFSET_CHECK usage.

Don't use CONFIG_FW_INFO_MAGIC_LEN.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

oyvindronningstad committed:
fw_info: Refactor EXT_API() macro

To make EXT_API code more readable.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

hakonfam committed:
cmake: move all logic from mcuboot/zephyr/CMakeLists.txt here

Do this to simplify working with multiple repositories.

Note that we can not use 'ZEPHYR_MCUBOOT_MODULE_DIR' as
this is not available in the modules included themself.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

trond-snekvik committed:
Bluetooth: Mesh: Light Lightness models

Adds the Light Lightness models:
- Light Lightness Client
- Light Lightness Server
- Light Lightness Setup Server

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

hakonfam committed:
cmake: include nrf reports cmake from repo

To make patches simpler and diff smaller.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

sigvartmh committed:
net: lib: aws_fota: Switch from Zephyr's JSON parser to cJSON

Switch from Zephyr's JSON parser to cJSON since Zephyr's JSON parser did
not handle nested JSON structures in a controllable manner. Where it
would either bus fault or copy random data into structure fields when
decoding structures with missing data.

Signed-off-by: Sigvart Hovland <sigvart.hovland@nordicsemi.no>

pdunaj committed:
applications: nrf_desktop: Porting guide

Jira:DESK-535

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>
Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>
Co-authored-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>
Co-Authored-By: Bartek Gentkowski <bartosz.gentkowski@nordicsemi.no>

pdunaj committed:
partition_manager: Allow passing location of static configuration

Jira:DESK-821

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

hakonfam committed:
cmake: move s1 patches to nrf repo

This to minimize patches and simplify development

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

pdunaj committed:
applications: nrf_desktop: Keyboard keep LED on only briefly

Jira:DESK-844

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

trond-snekvik committed:
Bluetooth: Added EnOcean library

Adds EnOcean library for observing BLE based EnOcean devices,
as well as a sample demonstrating basic functionality.

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

hakonfam committed:
manifest: update mcuboot

To get locking of mcuboot flash area before jump

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

plskeggs committed:
applications: asset_tracker: removed unused configs and fix wording

Removed left-behind CLOUD_UA_BUTTONS and CLOUD_UA_CONSOLE.  Fixed text
for CLOUD_BUTTON_INPUT to be more meaningful and not identical to
CLOUD_BUTTON.

Jira:TG91-172

Signed-off-by: Pete Skeggs <peter.skeggs@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Fix forwarding HID feature reports

Change fixes config channel state handling in HID forward module.
This removes the race condition.

Jira:DESK-847

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

oyvindronningstad committed:
fw_info: Fix checkpatch warnings

Add line after declaration and remove unneeded backslashes.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

plskeggs committed:
samples: nrf9160: lte_ble_gateway: Removed unused config CLOUD_UA_CONSOLE

Removed obsolete reference to CLOUD_UA_CONSOLE.

Jira:TG91-172

Signed-off-by: Pete Skeggs <peter.skeggs@nordicsemi.no>

jayteemo committed:
net: lib: nrf_cloud: Define and use return values for cloud_connect()

This will provide the user with useful information regarding
connect errors.

Signed-off-by: Justin Morton <justin.morton@nordicsemi.no>

jayteemo committed:
applications: asset_tracker: Improve cloud_connect() error handling

Use defined return values to improve error handling and user
notification.

Signed-off-by: Justin Morton <justin.morton@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Add passkey events

Change adds events used to request and provide passkey input.

Jira:DESK-830

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Add passkey input HW interface

Change adds passkey input HW interface. The implementation is based on
button events.

Jira:DESK-830

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Add Bluetooth passkey

Change adds Bluetooth passkey authentication to nRF Desktop keyboard.

Jira:DESK-830

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

oyvindronningstad committed:
fw_info: Add OPTIONAL config for EXT_API_REQs.

This allows adding optional EXT_API_REQs.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

oyvindronningstad committed:
bl_validation: Allow BL_VALIDATE_FW EXT_API_REQs to be optional.

Add a bl_validate_firmware_available() function for easily checking.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

MarekPieta committed:
scripts: profiler: Detect device family

Change adds detecting device family (device family is no longer
defined in configuration).

Jira:DESK-733

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
scripts: profiler: Use LowLevel API

Change switches the scripts to use LowLevel API (the currently used API
is becoming deprecated).

Jira:DESK-733

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

oyvindronningstad committed:
bl_crypto: cc310_hash: Reduce stack usage

Previous stack usage overflowed default stack size.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

oyvindronningstad committed:
secure_services: Cleanup

Whitespace, comments, and line length.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Add nrf52833_pca10100

Change adds nrf52833_pca10100 board configurations.
The board is used as nRF Destkop dongle.

Jira:DESK-804

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: CMakeLists.txt cleanup (events)

Cleanup the CMakeLists.txt for application events.

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Remove unnecessary ARG_UNUSED

Change removes unnecessary ARG_UNUSED in event source files.

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Use uart_fifo_fill only if declared

Change prevents ble_qos module from using uart_fifo_fill if
the function is undeclared.

Jira:DESK-804

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

martiles committed:
samples: nrf9160: http_application_update: startup message

Added a serial output of the specific sample you are running.

Signed-off-by: Martin Lesund <martin.lesund@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Fix clearing hid_forward state

Change fixes clearing hid_forward state on USB disconnection.

Jira:DESK-804

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
scripts: profiler: Fix pylint issues

Fix issues reported by pylint in rtt_nordic_profiler_host.py.

Jira:DESK-733

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

martiles committed:
samples: nrf9160: lte_ble_gateway: startup message

Added a more descriptive startup message.

Signed-off-by: Martin Lesund <martin.lesund@nordicsemi.no>

jayteemo committed:
net: lib: aws_iot: Use new return values for connect function

The cloud API's connect function now uses common return values
defined by enum cloud_connect_result.

Signed-off-by: Justin Morton <justin.morton@nordicsemi.no>

plskeggs committed:
application: asset_tracker: save power by triggering GPS update on motion

Add GPS_CONTROL_FIX_CHECK_OVERDUE to gps_controller/Kconfig; to be set
longer than *_INTERVAL value. Use GPS_START_ON_MOTION to conditionally
change gps reporting interval in gps_controller. Add
motion_activity_state_t to motion.h. Rewrite motion_handler() in main.c
to detect activity/inactivity state. Change structure of it to allow
triggering of GPS at end. Add new function motion_trigger_gps() to
trigger GPS no faster than shorter GPS_CONTROL_FIX_CHECK_INTERVAL on
motion, otherwise revert to new *_OVERDUE value to save power. Enhance
gps_controller.c to keep track of most recent time the GPS unit was
powered. Use most recent time powered in motion_trigger_gps() to ensure
GPS not triggered immediately on every movement (which hurts power
consumption).

Jira:TG91-7

Signed-off-by: Pete Skeggs <peter.skeggs@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Update sample.yaml

Change adds missing configurations to sample.yaml file.

Jira:DESK-804

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

erik-johnson-nimbelink committed:
lib: at_cmd_parser: Fixed warnings about char subscripts

Some toolchains will use macros that use the value as an array index,
so this combined with -Wchar-subscripts will give warnings about the
subscript being char (which tends to be used to remind people it's
potentially a signed value and might break out of arrays).

Signed-off-by: Erik Johnson <erik.johnson@nimbelink.com>

MarekPieta committed:
applications: nrf_desktop: Set proper initial connection parameters

Change sets proper initial connection parameters. This is workaround
required by nrfxlib Link Layer.

Jira:DESK-814

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

oyvindronningstad committed:
bootloader: provision: Add support for nRF53 app and net cores

app core: Place provision in OTP.
net core: Change provision size to match the flash page size in
          network core flash (2k).

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

evenl committed:
samples: nrf9160: gps: fixing compilation warnings

 * Fixing warning about implicit memset function declaration
 * Fixing warning about wrong argument when registering and
   deregistering at_notif handler

Signed-off-by: Even Falch-Larsen <even.falch-larsen@nordicsemi.no>

Rallare committed:
samples: nrf9160: http_application_update: Add bsdlib sys init handling

Add handling for when the application wants to boot up
with bsdlib_init called post-kernel, ie: with
CONFIG_BSD_LIBRARY_SYS_INIT=y
set in the project configuration.

Signed-off-by: Håkon Alseth <haal@nordicsemi.no>

oyvindronningstad committed:
tests: Fix bootloader/fw_info tests for nrf53.

Plus additional cleanup

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

erik-johnson-nimbelink committed:
lib: lwm2m_carrier: Added C++ guards

So it can be safely included.

Signed-off-by: Erik Johnson <erik.johnson@nimbelink.com>

erik-johnson-nimbelink committed:
lib: at_cmd: Added protecting against multiple initialization

There have been a few cases where multiple things have tried to
initialize the AT command module, only to have some bad things happen
(like the thread being created mutliple times and the slab allocated
for response handling permanently leaking at least one allocation).
This will at least make it easier for multiple things to call the
initialization routine without having that happen.

Signed-off-by: Erik Johnson <erik.johnson@nimbelink.com>

MarekPieta committed:
scripts: profiler: Cleanup - closing plot

Change cleans up closing plot.

Jira:DESK-733

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
scripts: profiler: Fix pylint issues in plot_nordic.py

Change fixes issues reported by pylint for plot_nordic.py file.

Jira:DESK-733

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

greg-fer committed:
doc: expand information about mesh models

NCSDK-3361: added details about common parts of models.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

kpuusaari committed:
lib: bsdlib: Add interpretation for NRF_EDOM

Added missing interpretation for NRF_EDOM.

Signed-off-by: Kimmo Puusaari <kimmo.puusaari@nordicsemi.no>

sigvartmh committed:
lib: bsdlib: Add Kconfig option to allow running BSDLib in secure

This allows a user for BSDLib to utilize the full chip and all it's
peripherals. The whole RAM is available except the block which is
occupied by statically allocating the size of memory needed and putting
it at the start of the location BSDLib expects it to be through the
`__attribute__` hint and linker script sections.

Signed-off-by: Sigvart Hovland <sigvart.hovland@nordicsemi.no>

sigvartmh committed:
samples: nrf9160: Simple BSDLib sample only from secure

This adds a sample displaying how you can use BSDLib from secure domain
only. Allowing the user of the application to use CryptoCell without
passing it through secure services. The user will also have full access
to RAM through Zephyr's code relocation API. Allowing you to place data,
text or bss before BSDLib's reserved memory. By default everything
compiled is placed after.

Signed-off-by: Sigvart Hovland <sigvart.hovland@nordicsemi.no>

sigvartmh commented:
Tested and working sample of running zephyr without TrustZone with BSDLib

oyvindronningstad commented on lib/bsdlib/bsdlib.c:
```diff
@@ -11,9 +11,79 @@
+
+static inline void setup_ns_power(void)
+{
+        NRF_SPU_S->PERIPHID[NRFX_PERIPHERAL_ID_GET(NRF_POWER_NS)].PERM = 0x80000102;
```
magic number

sigvartmh replied:
Replaced

oyvindronningstad commented on lib/bsdlib/bsdlib.c:
```diff
@@ -11,9 +11,79 @@
+
+static inline void setup_s_ipc(void)
+{
+        NRF_SPU_S->PERIPHID[NRFX_PERIPHERAL_ID_GET(NRF_IPC_S)].PERM = 0x80000102;
```
magic number

sigvartmh replied:
Replaced

oyvindronningstad commented on lib/bsdlib/bsdlib.c:
```diff
@@ -23,7 +93,7 @@ static int init_ret;
 {
 	/* Setup the network IRQ used by the BSD library.
-	 * Note: No call to irq_enable() here, that is done through bsd_init().
+//	 * Note: No call to irq_enable() here, that is done through bsd_init().
```
Remove slashes

sigvartmh replied:
Fixed.

oyvindronningstad commented on lib/bsdlib/bsdlib.ld:
```diff
@@ -0,0 +1,3 @@
+#include <bsd_platform.h>
+__bsdlib = BSD_RESERVED_MEMORY_ADDRESS;
```
I can't find where this is used.

oyvindronningstad replied:
If you want to place .bsdlib at BSD_RESERVED_MEMORY_ADDRESS, do

    . = BSD_RESERVED_MEMORY_ADDRESS;

sigvartmh replied:
I see that placing it wasn't the easiest thing to do. I can't seem to get the location right with `zephyr_linker_sources` I want to place the static allocated memory block into the start of . = BSD_RESERVED_MEMORY_ADDRESS but I don't know if it is possible. I kind of want to put it before kernel_data_end if possible. I think we discussed this before and that the linker script language is not powerful enough to express put Y between X and Z.

sigvartmh replied:
The version 62c7147 at least creates the section at the correct address. I'm a bit unsure about the implication of having it that way. Also, it seems like the `static u8_t __attribute__((section (".bsdlib"), unused)) bsd_mem[BSD_RESERVED_MEMORY_SIZE];
` is not placed at all. Doing `u8_t __attribute__((section (".bsdlib"))) bsd_mem[BSD_RESERVED_MEMORY_SIZE];` worked fine but I hoped to not have it exposed.

oyvindronningstad commented on samples/nrf9160/mqtt_simple_only_secure/src/main.c:
```diff
@@ -0,0 +1,434 @@
+#include <net/bsdlib.h>
+#include <bsd_platform.h>
+
+static u8_t __attribute__((section (".bsdlib"), unused)) bsd_mem[BSD_RESERVED_MEMORY_SIZE];
```
This is declared above as well. Is it meant to be declared twice?

sigvartmh replied:
No, forgot to remove this when I moved the other into the bsdlib source files.

SebastianBoe commented:
I'm not familiar enough with Trustzone, cryptocell, or bsd lib to be able to say if we want this or not. But the code generally looks clean and in general we don't want to limit what users can do more than necessary so it sounds sane.

sigvartmh commented:
@SebastianBoe Seems I currently have an issue with the placement of the unused/noinit section for the BSDLib memory section in the linker script. I really wished I was able to come up with a way of giving users access to the 64K of RAM before the BSD section and 256K - 49K - 64K after the BSD section.

Do you know a way of expressing this through the linker?

MarekPieta committed:
scripts: profiler: Keep reference to FuncAnimation object

Keep reference to FuncAnimation object to ensure that it will not
be garbage collected.

Jira:DESK-733

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

rlubos committed:
lib: bsdlib: Fix accept address handling

bsdlib expects to receive specific sizes in `p_addrlen` for IPv4 and
IPv6, so we need to add extra address handling when translating from nrf
types to Zephyr types.

Also, add NULL pointer handling.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

Commit:
doc: add link to Thingy:91 User Guide

Add a link to the Thingy User Guide to the samples that
can be run on Thingy:91.

Signed-off-by: Ruth Fuchss <ruth.fuchss@nordicsemi.no>

hakonfam commented on lib/bsdlib/Kconfig:
```diff
@@ -5,13 +5,17 @@
 
+config BSD_LIBRARY_ALLOW_SECURE
+	bool "Allow usage of BSD library in secure(Experimental)"
+	default n
```
IMO we can skip this as it is the default

sigvartmh replied:
Skip the whole option? We need something to notify BSDLib that you are now going to allow running it without TrustZone(the user knows what he's doing) option.

If not we could just remove the TrustZone requirement but I'm afraid we are creating a foot gun by allowing users who don't understand the implications of doing this that their RAM will be split etc.

hakonfam replied:
No not the option, just the `default n` part

hakonfam commented on lib/bsdlib/bsdlib.c:
```diff
@@ -42,8 +121,12 @@ static int _bsdlib_init(struct device *unused)
 int bsdlib_init(void)
 {
+#ifdef CONFIG_BSD_LIBRARY_ALLOW_SECURE
+	setup_for_secure();
```
Curious, why is this needed?

hakonfam replied:
If its more then 1 line just explain it IRL xD

sigvartmh replied:
It's needed to setup the clock(BSDLib tries to enable the LF clock if it's not enabled) the modem is similar to nRF53 where it requires the RAM section used to be setup as non-secure(if not it won't be addressable on the BUS since the Modem is always a non-secure domain) Same with power it needs to be set as NS since BSDLib uses NS addressing same with the IPC. I think maybe we could allow POWER to be split. Sorry for the magic values seems they got reverted at some point.

So basically:
IPC NS
POWER NS or SPLIT
RAM NS

Are the requirements for operating with the modem from the secure domain.

We could also make
CLOCK NS and maybe it would work then we don't have to enable the LF clock and BSDLib could maybe handle it correctly if we allow Non-secure mapping. I'll have to test that.

hakonfam commented on samples/nrf9160/mqtt_simple_only_secure/CMakeLists.txt:
```diff
@@ -0,0 +1,9 @@
+project(hello_world)
+
+zephyr_linker_sources(SECTIONS app-data-section.ld)
+target_sources(app PRIVATE src/main.c)
```
Instead of copying the entire source dir, could you simply point to the source in the original sample folder, and then have the separate prj.conf in this folder as well. Assuming that there are no code chanes.

sigvartmh replied:
The sample file is mostly for testing and just ensuring that it actually works I think I would remove it in favor of document how to build your application. So by using the `__in_section` we should be able to place whatever we prefer in the split RAM. Using https://docs.zephyrproject.org/latest/guides/code-relocation.html will actually allow you to place .bss/.text/.data sections in the split RAM also. So if you fill up your first 64K you should be able to place additional sections into `SRAM_AFTER_BSDLIB` 128K

mbolivar-nordic committed:
doc: add a note about asset_tracker boot chain

I personally lost some time doing some development related to this
application due to an incorrect assumption that it was using the
secure bootloader chain which is documented in ug_bootloader.rst.

In fact, by default, this application does not use the secure
bootloader chain.

In case others are confused by this, try to help by adding a short
paragraph explaining this.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>
Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>
Signed-off-by: Ruth Fuchss <ruth.fuchss@nordicsemi.no>

umapraseeda committed:
doc: Created subpages for libraries

Created subpages for libraries to improve the structure.
ref:https://projecttools.nordicsemi.no/jira/browse/NCSDK-4349

Signed-off-by: Uma Praseeda <uma.praseeda@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Add HID system control event

Change adds HID system control event.

Jira:DESK-639

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Add system control HID report description

Change adds system control HID report description.

Jira:DESK-639

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
applications: nrf_destkop: Add system control support (peripheral)

Change adds HID system control report support on peripheral devices.

Jira:DESK-639

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Add system control support (central)

Change adds HID system control report support on central devices.

Jira:DESK-639

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Add HID system report to configuration

Change adds using HID system reports as part of default configuration
for pca20037 and pca10059.

Jira:DESK-639

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

umapraseeda committed:
doc: Added information about LTE Link Monitor in getting started

Added missing information about connecting to nRF9160 devices
using LTE Link Monitor in getting started(testing) section.
Moved a relevant general section from AT Client sample to the
testing section.
ref:https://projecttools.nordicsemi.no/jira/browse/NCSDK-1104

Signed-off-by: Uma Praseeda <uma.praseeda@nordicsemi.no>

rlubos committed:
west.yml: v1.2.0 release

Updated the west.yml to use release tags:
- zephyr:  v2.1.99-ncs1
- mcuboot: v1.4.99-ncs2
- nrfxlib: v1.2.0
- mcumgr: v0.0.1

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Whitelist scan requests

Change adds using whitelist to filter devices that can request scan
response data.

Jira:DESK-857

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

Commit:
doc: update link to SUPL client

The SUPL client library is now available on the website.
Update the link to point directly to the library.

Signed-off-by: Ruth Fuchss <ruth.fuchss@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Switch to name filters

Change switches to name filters on nRF Desktop dongles.
The dongles will no longer try to connect with Peripherals bonded
with another host.

Jira:DESK-857

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

oyvindronningstad committed:
bl_crypto: Make it possible to disable SHA256 and SECP256R1

If using the CLIENT backend, force EXT_API_REQs to be required.
The new NO backend allows bl_crypto and therefore fw_info to be included
without requiring EXT_APIs or including crypto code.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

Commit:
doc: add link to security tag information

Add link that was commented out before (because the target
was not available yet).

Signed-off-by: Ruth Fuchss <ruth.fuchss@nordicsemi.no>

erik-johnson-nimbelink committed:
lib: at_notif: Added protecting against multiple initialization

Otherwise it might end up dropping handlers that have already been
registered.

Signed-off-by: Erik Johnson <erik.johnson@nimbelink.com>

pdunaj committed:
applications: nrf_desktop: Disable serial driver for DK HW config

By default DK HW enables serial console which is by default not
used in our project.
For release configurations we also need to make sure that Segger RTT
is disabled.

Jira:DESK-851

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

pdunaj committed:
boards: Add PCA10111 definition files

PCA10111 is a nrf52833 based USB dongle.

Jira:DESK-760

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

kapi-no committed:
doc: nrf5340_ug: flow control description for logging on net core

Covered the description of connections for flow control lines in the
network core logging section.

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

tommi-m committed:
lib: bsdlib: Fix error codes in getaddrinfo

Added conversion from nrf error codes to zephyr
error codes.

Signed-off-by: Tommi Mammela <tommi.mammela@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Add missing LL related ifdefs

Change adds missing ifdefs related to LL selection.
This is required to switch LL on Central.

Jira:DESK-856

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

greg-fer committed:
doc: add link to MQTT simple samples

Added References section with link to a devzone tutorial.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

b-gent committed:
doc: update versions after release

Update ncs and nrfxlib versions to 1.2.99.

Signed-off-by: Bartosz Gentkowski <bartosz.gentkowski@nordicsemi.no>

oyvindronningstad committed:
spm: secure_services: Change default for PREVALIDATE to 'n'

SPM doesn't boot with this if B0 is not present.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

oyvindronningstad committed:
tests: bl_crypto: Fix tests

Remove unneeded line in CMakeLists.txt and change which EXT_API
kconfig is enabled in prj.conf.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

pdunaj committed:
boards: Add PCA10114 definition files

PCA10114 is nrf52833 based dongle with a limited memory.

Jira:DESK-760

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Increase number of supported HID reports

Change increases number of HID report supported by HID client. This
ensures that dongle will be able to subscribe to every HID report.

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

oyvindronningstad committed:
tests: fw_info: Remove the need to specify the provision hex.

Check BOARD definition for chip names

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

pdunaj committed:
applications: nrf_desktop: Use static partitions configuration

To ensure that partition layout does not change between builds
static configuration of partitions must be used.

If partition layout changes DFU can lead to mismatch in position
of settings partition and so loss of persistent data like
BLE bonds or user settings.

Jira:DESK-821

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

pdunaj committed:
applications: nrf_desktop: Add configuration for PCA10111

Jira:DESK-760

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

pdunaj committed:
boards: nrf52833_pca10111: Boost GPIO voltage

By default nrf52833 will use 1.8V external voltage if powered from
USB bus. This change make it use 3.0V instead.

Jira:DESK-760

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

pdunaj committed:
applications: nrf_desktop: Fix typo in config files

Commit fixes typo in comments in one of the config files.

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

Commit:
doc: update mcuboot location

MCUboot has moved from mcuboot to bootloader/mcuboot.
Reflect this change in the documentation.

Signed-off-by: Ruth Fuchss <ruth.fuchss@nordicsemi.no>

martiles committed:
applications: asset_tracker: Option for NB-IoT_GPS

Adding the option for selecting NB-IoT and GPS easily
in the prj.conf.

Signed-off-by: Martin Lesund <martin.lesund@nordicsemi.no>

thst-nordic committed:
ci: [temp-hack] pip install pylint

it is missing from ci-tools requirements

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>

plskeggs committed:
application: asset_tracker: fix formatting of Kconfig help text

During code review it was noticed that many of the Kconfig help text
sections in this file do not follow the indenting standard of one
tab plus two spaces.  This commit makes them all follow the rule.

Signed-off-by: Pete Skeggs <peter.skeggs@nordicsemi.no>

junqingzou committed:
Sample: serial_lte_modem add AT#XCLAC and AT#XPING

Add AT#XCLAC list all AT commands in SLM (add-on to AT+CLAC)
Add AT#XPING implement ICMP EchoRequest with nRF raw socket
Remove the support of UART_1, which is reserved for modem trace
Remove unnecessary CONFIG flags

Signed-off-by: Jun Qing Zou <jun.qing.zou@nordicsemi.no>

pdunaj committed:
applications: nrf_desktop: Move porting guide to readme file

Move porting instructions from a separate page to the main
readme file of nrf_desktop.

Jira:DESK-535

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>
Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>
Co-authored-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

kapi-no committed:
samples: peripheral: radio_test: align use of clock_control API

Aligned the Radio Test sample with the changes to the Clock Control
API.

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Limit discoverability if bonded

Change removes general discoverable from Bluetooth advertising packet
in case Peripheral is bonded.

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

oyvindronningstad committed:
samples: secure_services: Print FICR.INFO.VARIANT

In addition to FICR.INFO.PART.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

oyvindronningstad committed:
samples: secure_services: Allow services to be disabled.

Since some services are not always available.
Improve newline printing so printout looks ok when services are
disabled.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

MarekPieta committed:
scripts: hid_configurator: Update report ID

Change updates report ID used by hid_configurator.
The report ID must match report ID on connected HID device.

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
scripts: hid_configurator: Fix pylint issues

Change fixes issues reported by pylint.

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

plskeggs committed:
application: asset_tracker: make FOTA active again

Commit 0516f79f requires a new config value
CONFIG_FOTA_DOWNLOAD=y in order for FOTA to function.

Signed-off-by: Pete Skeggs <peter.skeggs@nordicsemi.no>
